File: Atcoder/abc057/c.cpp
#include <bits/stdc++.h>
using namespace std;
#define rep(i,N) for(int i=0;i<int(N);++i)
#define rep1(i,N) for(int i=1;i<int(N);++i)

typedef long long ll;

int cnt_digits(ll X){
	int cnt=0;
	while(X>0){
		X/=10;
		cnt++;
	}
	return cnt;
}

int main() {
	ll N;
	cin>>N;
	int ans=cnt_digits(N);
	rep1(A,1e5+1){
		if(N%A==0){
			ll B=N/A;
			int C=max(cnt_digits(A),cnt_digits(B));
			ans=min(ans,C);
		}
	}
	cout<<ans<<endl;
}

File: atcoder/abc057/c.py
N = int(input())

root_N = pow(N,1/2)

ans = 1000
for i in range(int(root_N)+1):
	A = i+1
	if N%A == 0:
		B = N//A
		ans_l = max(len(str(A)), len(str(B)))
		ans = min(ans,ans_l)
print(ans)